loopFreq : 0.5
